### PR TITLE
Multi-leg nearby coordinate lookups.

### LINKED ISSUE #1879: nearbyCoordinates does not look for all nearby coordinates
Currently we get coordinates that are nearby a step (step before, current step, step after) from `RouteLegProgress`, those coordinates are important for course following calculations.

The problem is that in certain circumstances (starting a new leg, specifically), those `nearbyCoordinates` do not have the last step of the previous leg, which can, in certain situations, cause reroutes, as the course-detection logic does not have a complete picture of the maneuver geometry.

The solution here is to move `nearbyCoordinates` to `RouteProgress`, and do a lookback/lookahead when we're either on the first step of a new leg, or the last step of a new leg, respectively.

/cc @mapbox/navigation-ios

### PR DESCRIPTION
**Pull request by JThramer**
Fixes #1879. Allows for look-behind when you're on the first step of a leg, and look-ahead when you're on the last step.

/cc @mapbox/navigation-ios

### CODE REVIEW

File: MapboxCoreNavigation/RouteController.swift [hunk 1]
```diff
@@ -221,13 +221,7 @@ open class RouteController: NSObject, Router {
...
      */
     func userCourseIsOnRoute(_ location: CLLocation) -> Bool {
-        // if we have yet to travel along the current leg, don't check for heading conformance
-        guard routeProgress.currentLegProgress.distanceTraveled > 0 else {
-            movementsAwayFromRoute = 0
-            return true
-        }
-        
-        let nearByCoordinates = routeProgress.currentLegProgress.nearbyCoordinates
+        let nearByCoordinates = routeProgress.nearbyCoordinates
```
Comment (akitchen) [inline MapboxCoreNavigation/RouteController.swift]: Does it make sense to keep the currentLegProgress.nearbyCoordinates in addition to the new routeProgress.nearbyCoordinates? should we be deprecating the older one?

also, the B in nearby should be a small b, but that's ultra-minor

Comment (JThramer) [reply MapboxCoreNavigation/RouteController.swift]: I've no real feeling on the subject either way, but @1ec5 did argue for keeping it the other day, and his rationale (which I cannot remember OOTOMH) made sense to me.

File: MapboxCoreNavigation/RouteProgress.swift [hunk 2]
```diff
@@ -98,6 +98,42 @@ open class RouteProgress: NSObject {
...
+        //If we're on the last step of a leg, add coordinates from the next leg.
+        if let lastStep = currentLegProgress.leg.steps.last,
+            currentLegProgress.currentStep == lastStep,
+            legIndex < (route.legs.count - 1) {
+            
+            let nextLeg = route.legs[legIndex + 1]
+            
+            if let nextStep = nextLeg.steps.first,
+               let upcomingCoords = nextStep.coordinates {
+                coords += upcomingCoords
```
[the remark this answers is not in the document]
Comment (1ec5) [reply MapboxCoreNavigation/RouteProgress.swift]: Before acting on this feedback, please double-check my assumption that the coordinates would be duplicated. Off the top of my head, I’m unsure whether that’s true in the case where one or both legs’ coordinates for the same waypoint get snapped differently (for example due to different approach settings).

[the remark this answers is not in the document]
Comment (JThramer) [reply MapboxCoreNavigation/RouteProgress.swift]: Duplication confirmed.
```
(lldb) po currentLegProgress.upcomingStep!.coordinates?.last
▿ Optional<CLLocationCoordinate2D>
  ▿ some : CLLocationCoordinate2D
    - latitude : 39.752338
    - longitude : -105.001484

(lldb) po upcomingLeg?.steps[0].coordinates?.first
▿ Optional<CLLocationCoordinate2D>
  ▿ some : CLLocationCoordinate2D
    - latitude : 39.752338
    - longitude : -105.001484
```

File: MapboxCoreNavigation/RouteProgress.swift [hunk 3]
```diff
@@ -364,18 +414,6 @@ open class RouteLegProgress: NSObject {
         currentStepProgress = RouteStepProgress(step: leg.steps[stepIndex], spokenInstructionIndex: spokenInstructionIndex)
     }
 
-    /**
-     Returns an array of `CLLocationCoordinate2D` of the prior, current and upcoming step geometry.
-     */
-    @objc public var nearbyCoordinates: [CLLocationCoordinate2D] {
```
[the remark this answers is not in the document]
Comment (akitchen) [reply MapboxCoreNavigation/RouteProgress.swift]: I think we should do a formal deprecation here, point usages to the new API, and delete in an upcoming release. can do?

**Pull request merged**